### PR TITLE
Adds pg 16 support

### PR DESCRIPTION
**Pull request by gurkanindibay**
Here Pg 16 Build
https://github.com/citusdata/packaging/actions/runs/5946975324/job/16128668542#step:8:2065

Manual test results:

Ubuntu focal
```bash

postgres@5a4442384a6f:~$ psql -p 9700 -c "select citus_version();"
                                                       citus_version                                                        
----------------------------------------------------------------------------------------------------------------------------
 Citus 12.1devel+git.20230823.36b51d6 on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0, 64-bit
(1 row)

postgres@5a4442384a6f:~$ psql -p 9700 -c "select version();"
                                                                 version                                                                  
------------------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 16beta3 (Ubuntu 16~beta3-1.pgdg20.04+2) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0, 64-bit
(1 row)
```
el/8

```bash
-bash-4.2$ psql -p 9700 -c "CREATE EXTENSION citus;"
CREATE EXTENSION
-bash-4.2$ psql -p 9700 -c "select citus_version();"
                                                        citus_version                                                         
------------------------------------------------------------------------------------------------------------------------------
 Citus 12.1devel+git.20230821.ba2a0ae on x86_64-pc-linux-gnu, compiled by gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-18), 64-bit
(1 row)

-bash-4.2$ psql -p 9700 -c "select version();"
                                                  version                                                   
------------------------------------------------------------------------------------------------------------
 PostgreSQL 16beta3 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44), 64-bit
(1 row)

-bash-4.2$
```

### CODE REVIEW

File: .github/workflows/image-health-check.yml [hunk 1]
```diff
@@ -16,9 +16,14 @@ on:
     branches: "**"
   schedule:
     - cron: "30 2 * * *"
-
   workflow_dispatch:
 
+concurrency:
```
Comment (gurkanindibay) [inline .github/workflows/image-health-check.yml:21]: This configuration is to cancel the previous pipeline execution 
https://docs.github.com/en/actions/using-jobs/using-concurrency

https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre

File: .github/workflows/build-citus-community-nightlies.yml [hunk 2]
```diff
@@ -17,6 +17,11 @@ on:
     - cron: "30 1 * * *"
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```
Comment (hanefi) [inline .github/workflows/build-citus-community-nightlies.yml]: How about using a simpler expression that only contains workflow and git reference name?

```suggestion
  group: ${{ github.workflow }}-${{ github.ref }}
```

This is the suggested approach in the documentation [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow).

File: .github/workflows/build-citus-community-nightlies.yml [hunk 3]
```diff
@@ -17,6 +17,11 @@ on:
...
+  cancel-in-progress: true
+
+
```
Comment (hanefi) [inline .github/workflows/build-citus-community-nightlies.yml]:
```suggestion
```

File: .github/workflows/build-package-test.yml [hunk 4]
```diff
@@ -15,6 +15,9 @@ on:
   push:
     branches: "**"
   workflow_dispatch:
+concurrency:
```
Comment (hanefi) [inline .github/workflows/build-package-test.yml:19]:
```suggestion

concurrency:
```

File: .github/workflows/build-package.yml [hunk 5]
```diff
@@ -14,9 +14,14 @@ env:
...
     branches: "**"
-
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+
+
```
Comment (hanefi) [inline .github/workflows/build-package.yml]:
```suggestion
```

Please remove extra empty lines. Having one empty line is the best choice IMO.

File: .github/workflows/build-citus-community-nightlies.yml (hunk 2 above)
Comment (hanefi) [reply .github/workflows/build-citus-community-nightlies.yml]: Note: This comment is valid for all the concurrency.group attributes introduced in this PR.

File: .github/workflows/image-health-check.yml [hunk 6]
```diff
@@ -16,9 +16,14 @@ on:
...
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+
+
```
Comment (hanefi) [inline .github/workflows/image-health-check.yml]: too many empty lines.

File: templates/Dockerfile-rpm.tmpl [hunk 7]
```diff
@@ -75,6 +75,7 @@ RUN if [[ %%os%%   == oraclelinux ]] && [[ %%release%% == 7 ]]; then yum install
 
 # install build tools and PostgreSQL development files
 RUN ( yum install -y https://%%rpm_url%%  \
+    && ( [[ 16 != 16 ]] || sed -i '/\[pgdg16-updates-testing\]/{n;n;n;s/.*/enabled=1/}' /etc/yum.repos.d/pgdg-redhat-all.repo ) \
```
Comment (hanefi) [inline templates/Dockerfile-rpm.tmpl]: I do not think that `[[ 16 != 16 ]]` part is correct. I guess you tried to do a comparison between 16 and a variable that stores the PG version in there.

File: .github/workflows/build-package-test.yml (hunk 4 above)
Comment (gurkanindibay) [reply .github/workflows/build-package-test.yml:19]: Fixed

File: .github/workflows/build-citus-community-nightlies.yml (hunk 2 above)
Comment (gurkanindibay) [reply .github/workflows/build-citus-community-nightlies.yml]: Changed for all pipelines

File: .github/workflows/build-citus-community-nightlies.yml (hunk 3 above)
Comment (gurkanindibay) [reply .github/workflows/build-citus-community-nightlies.yml]: Fixed

File: .github/workflows/build-package.yml (hunk 5 above)
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: Fixed

File: .github/workflows/image-health-check.yml (hunk 6 above)
Comment (gurkanindibay) [reply .github/workflows/image-health-check.yml]: Fixed

File: templates/Dockerfile-rpm.tmpl (hunk 7 above)
Comment (gurkanindibay) [reply templates/Dockerfile-rpm.tmpl]: Good catch. Fixed this one as well

**Review by hanefi — APPROVED** [review]

**Pull request merged**